### DISCUSSION TITLE
Recommended way to x-for with dynamic content from Livewire component

### DISCUSSION THREAD

**Discussion by waterbuckit** [alpinejs/alpine#902] [1. Help]
I have a component which has an x-for in it, and the list is populated as such:
```
<ul>
    <li x-show="adding">
    </li>
    <template x-for="(item, index) in {{json_encode(array_values($items))}}" :key="index">
        <li>
        ...
        </li>
    </template>
</ul>
```
However, when I add a new element to `items`, the Livewire component refreshes and I am left with an empty list, even though the array is certainly filled in there as it should be. What is the recommended way to handle this? I've been running into all sorts of trouble with multiple ways of implementing this kind of thing (tried blade directives instead of alpine for-loops too). I have my x-data defined in a separate component, which could be a part of the problem, but I need access to properties from that with elements in my list. I need that special `li` at the top for when a user is adding an element, it's temporarily shown until they add it, yet, unfortunately because of these problems it becomes shown on refresh. As though it just ignores the fact that "adding" is false at this point.

Many thanks.

**Comment by KevinBatdorf** [comment]
I think you need to use `@entangle` in `x-data`, right? Something like:

```html
<ul x-data="{ myItems: @entangle('items') }">
    <li x-show="adding">
    </li>
    <template x-for="(item, index) in Object.values(myItems)" :key="index">
        <li>
        ...
        </li>
    </template>
</ul>

```

https://laravel-livewire.com/docs/2.x/alpine-js
Scroll down to the `@entangle` part

If not, someone else here might be able to help, or on the Livewire repo/Discord
